src/core/mac/mac.cpp

```diff
@@ -1637,6 +1637,12 @@ otError Mac::ProcessReceiveSecurity(RxFrame &aFrame, const Address &aSrcAddr, Ne
         }
 
         aNeighbor->SetLinkFrameCounter(frameCounter + 1);
+#if OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
+        if (aNeighbor->GetLinkFrameCounter() > aNeighbor->GetLinkAckFrameCounter())
```

Comment:
How about the follwoing: (I am thinking of this to help reduce the potenrtial conflicts with the multi-radio code and its  rebase if/when this is merged.).
```suggestion
        if (frameCounter + 1 > aNeighbor->GetLinkAckFrameCounter())
```